py17track/profile.py

```
@@ -108,3 +109,70 @@ async def summary(self, show_archived: bool = False) -> dict:
         for kind in summary_resp.get("Json", {}).get("eitem", []):
             results[PACKAGE_STATUS_MAP[kind["e"]]] = kind["ec"]
         return results
+
+    async def add_package(
+        self, tracking_number: str, friendly_name: Optional[str] = None
+    ) -> bool:
+        """Add a package by tracking number to the tracking list."""
+        add_resp: dict = await self._request(
+            "post",
+            API_URL_BUYER,
+            json={
+                "version": "1.0",
+                "method": "AddTrackNo",
+                "param": {"TrackNos": [tracking_number]},
+            },
+        )
+
+        _LOGGER.debug("Add package response: %s", add_resp)
+
+        if add_resp.get("Code") != 0:
+            return False
+
+        if friendly_name is not None:
+            packages = await self.packages()
+            new_package = next(
+                filter(lambda p: (p.tracking_number == tracking_number), packages), None
+            )
+
+            if new_package is not None:
+                _LOGGER.debug(
+                    "Found internal ID of recently added package: %s", new_package.id
+                )
+
+                success = await self.set_friendly_name(new_package.id, friendly_name)
+                if success is False:
```

Don't swallow errors.